x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/components/indices_select_combobox.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useEffect } from 'react';
+
+import { useValues, useActions } from 'kea';
+import useThrottle from 'react-use/lib/useThrottle';
+
+import {
+  EuiComboBox,
+  EuiComboBoxProps,
+  EuiComboBoxOptionOption,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHealth,
+  EuiHighlight,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+import { FormattedNumber } from '@kbn/i18n-react';
+
+import { Status } from '../../../../../../common/types/api';
+import { ElasticsearchIndexWithIngestion } from '../../../../../../common/types/indices';
+
+import { indexHealthToHealthColor } from '../../../../shared/indices/utils';
+import { FetchIndicesForEnginesAPILogic } from '../../../api/engines/fetch_indices_api_logic';
+
+export type IndicesSelectComboBoxProps = Omit<
+  EuiComboBoxProps<ElasticsearchIndexWithIngestion>,
+  'onCreateOption' | 'onSearchChange' | 'noSuggestions' | 'async'
+> & {
+  'data-telemetry-id'?: string;
+};
+
+const ENGINES_SEARCH_THROTTLE_MS = 500;
+
+export const IndicesSelectComboBox = (props: IndicesSelectComboBoxProps) => {
+  const [searchQuery, setSearchQuery] = useState<string | undefined>(undefined);
+  const throttledSearchQuery = useThrottle(searchQuery, ENGINES_SEARCH_THROTTLE_MS);
```

Comment:
question: does it make sense to use a [kea breakpoint](https://keajs.org/docs/core/listeners/#breakpoints) for this (could even add the timeout size to the createApiLogic as an option so we have access to this across the codebase)?